### PR TITLE
add ability to instance with modifications

### PR DESCRIPTION
**Pull request by hanbollar**

### CODE REVIEW

**Comment by hanbollar** [comment]
pausing the work here to do directly on Laurent's demo proj and then i'll copy it back over to here once worked through since the demo is more priority

https://github.com/lobau/lastquestion

**Comment by hanbollar** [comment]
moving back over from lastquestion starting with https://github.com/Volumetrics-io/MR.js/pull/144/commits/c51e465e72ab6a4f1a3a7f52036455d57749bb92

**Comment by michaelthatsit** [comment]
@hanbollar is this ready for review?

File: src/component-systems/InstancingSystem.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+        // grab usable mesh
+        if (originalMesh instanceof THREE.Mesh) {
+            combinedGeometry = originalMesh.geometry.clone();
+        } else if (originalMesh instanceof THREE.Group) {
+            originalMesh.traverse((child) => {
+                if (child instanceof THREE.Mesh) {
+                    const geometry = child.geometry.clone();
+                    geometry.applyMatrix4(child.matrixWorld); // Apply the child's world matrix
+                    combinedGeometry.merge(geometry);
```
[the remark this answers is not in the document]
Comment (hanbollar) [reply src/component-systems/InstancingSystem.js:55]: `mergeGeometries` would be a static function on an array of geometries from different instances and return the singular merged geometry of it - whereas `merge` here is used by the buffer geometry to merge in an additional geometry to it

**Pull request merged**